project_duplicate_subtask/models/project_task.py

```diff
@@ -1,15 +1,27 @@
 # Copyright (C) 2021 ForgeFlow S.L.
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
 
-from odoo import models
+from odoo import api, models
 
 
 class ProjectTask(models.Model):
     _inherit = "project.task"
 
+    @api.model
+    def _extract_action_dict(self, action):
+        action = action.sudo()
+        return {
+            "name": action.name,
+            "res_model": action.res_model,
+            "view_mode": action.view_mode,
+            "context": action.context,
+            "domain": action.domain,
+            "search_view_id": action.search_view_id,
+            "help": action.help,
+        }
+
     def action_duplicate_subtasks(self):
-        action = self.env.ref("project.action_view_task")
-        result = action.read()[0]
+        action = self._extract_action_dict(self.env.ref("project.action_view_task"))
```

Comment:
You can just make self as sudo() and it will allow to duplicate for all users. I don't think you need to fix in other parts of module.
```suggestion
        self = self.sudo()
        action = self.env.ref("project.action_view_task")
        result = action.read()[0]
```

Reply:
@AungKoKoLin1997 Nope. You should use _for_xml_id() function. Search in code to have example.